app/controllers/proposals_controller.rb

```diff
@@ -148,4 +149,28 @@ def require_waitlisted_or_accepted_state
       redirect_to event_url(@event.slug)
     end
   end
+
+  def require_complete_profile
+    return if current_user.complete? && current_user.unconfirmed_email.blank?
+
+    @incomplete_profile_msg = construct_incomplete_profile_msg.html_safe
```

Comment:
yup...I did not rework this after switching the overall approach to flash a warning/danger message. It does seem better inline, and also good to know that avoiding ivars in controllers is a preferred approach.